include/deal.II/numerics/vector_tools_constraints.templates.h

```diff
@@ -529,26 +670,29 @@ namespace VectorTools
                     internal::VectorDoFTuple<dim> vector_dofs;
                     vector_dofs.dof_indices[0] = face_dofs[i];
 
+                    internal::FaceDoFInfo face_dof_info;
+                    face_dof_info.indices[0] = cell->active_fe_index();
+                    face_dof_info.indices[1] = face_no;
+                    face_dof_info.indices[2] = i;
+
+                    std::array<unsigned int, dim> local_vector_indices;
+                    for (unsigned int d = 0; d < dim; ++d)
+                      local_vector_indices[d] = numbers::invalid_unsigned_int;
+
+                    const auto it = dof_to_vector_dof.find(face_dof_info);
+                    if (it != dof_to_vector_dof.end())
```

Comment:
should this be an error if ``it==end()``?

Reply:
if `it==end()`, then `local_vector_indices` would be `invalid_unsigned_int` since no key has been found.

Reply:
Is this situation supposed to happen during a normal computation?

Reply:
Yes, this is how `dof_to_vector_dof` is computed.

Reply:
I don't understand. What does the code below do if you can not find it in the map and the value is ``numbers::invalid_unsigned_int``?